doc/source/user-storage.md

```diff
@@ -0,0 +1,207 @@
+detect that the user has a pre-existing `PVC` and will simply
+attach it to their new pod, rather than creating a new `PVC`.
+
+### How can this process break down?
+
+When Kubernetes uses the `PVC` to create a new user`PV`, it
+is sending a command to the underlying API of whatever cloud
+provider Kubernetes is running on. Occasionally, the request
+for a specific `PV` might fail - for example, if your account
+has reached the limit in the amount of disk space available.
+Another common issue is limits on the number of volumes that
+may be simultaneously attached to a node in your cluster. Check
+your cloud provider for details on the limits of storage
+resources you request.
+
+## Configuration
+
+Most configuration for storage is done at the cluster level and
+is not unique to JupyterHub. However, some bits are, and we will
+demonstrate here how to configure those.
+
+Note that new `PVC`s for pre-existing users will **not** be
+created unless the old ones are destroyed. If you update your
+users' `PVC`  config via `config.yaml`, then any **new** users will
+have the new `PVC` created for them, but **old** users will not.
+To force an upgrade of the storage type for old users, you will
+need to manually delete their `PVC` (e.g.
+`kubectl --namespace=<your-namespace> delete pvc <pvc-name>`).
+**This will delete all of the user's data** so we recommend
+backing up their filesystem first if you want to retain their data.
+
+After you delete the user's `PVC`, upon their next log-in a new
+`PVC` will be created for them according to your updated `PVC`
+specification.
+
+### Type of storage provisioned
+
+A [StorageClass](https://kubernetes.io/docs/concepts/storage/storage-classes/) object
+is used to determine what kind of `PersistentVolume`s are provisioned for your
+users. Most popular cloud providers have a `StorageClass` marked as default. You
+can find out your default `StorageClass` by doing:
+
+```bash
+kubectl get storageclass
+```
+
+and looking for the object with `(default)` next to its name.
+
+To change the kind of `PersistentVolume`s provisioned for your users,
+
+1. Create a new `StorageClass` object following the
+   [kubernetes documentation](https://kubernetes.io/docs/concepts/storage/storage-classes/)
+2. Specify the name of the `StorageClass` you just created in `config.yaml`
+   ```yaml
+   singleuser:
+     storage:
+       dynamic:
+         storageClass: <storageclass-name>
+   ```
+3. Do a `helm upgrade`
+
+Note that this will only affect new users who are logging in. We recommend
+you do this before users start heavily using your cluster.
+
+We will provide examples for popular cloud providers here, but will generally
+defer to the Kubernetes documentation.
+
+#### Google Cloud
+
+On Google Cloud, the default `StorageClass` will provision
+Standard [Google Persistent Disk](https://cloud.google.com/compute/docs/disks/#pdspecs)s.
+These run on Hard Disks, and are not very fast. If you want to use SSDs, you
```

Comment:
Hard Disks ~~, and are not very fast~~ For more performance, you may want to use SSDs. For SSDs, you can create...